sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -191,9 +214,25 @@ protected override void Dispose(bool disposing)
             }
         }
 
-        private static long GetBlobLength(Response<IDownloadedContent> response)
+        private async Task<long> GetBlobLengthInternal(bool async, CancellationToken cancellationToken)
+        {
+#pragma warning disable AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+            Response<TProperties> response = await _getPropertiesInternalFunc(async, cancellationToken).ConfigureAwait(false);
+#pragma warning restore AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+
+            response.GetRawResponse().Headers.TryGetValue("Content-Length", out string lengthString);
+
+            if (lengthString == null)
+            {
+                throw new ArgumentException("Content-Range header is mssing on get properties response.");
```

Comment:
The headers don't match.  Consider using constants like `HttpHeader.Names.ContentLength` instead.

Reply:
Fixed in next iteration.